alt_train.py

```diff
@@ -0,0 +1,158 @@
+                                               batch_size=batch_size, shuffle=True,
+                                               num_workers=0, pin_memory=True)
+    evaluator = Evaluator(path_to_val_lmdb_dir)
+    optimizer = optim.SGD(model.parameters(), lr=initial_learning_rate, momentum=0.9, weight_decay=0.0005)
+    scheduler = StepLR(optimizer, step_size=training_options['decay_steps'], gamma=training_options['decay_rate'])
+
+    if path_to_restore_checkpoint_file is not None:
+        assert os.path.isfile(path_to_restore_checkpoint_file), '%s not found' % path_to_restore_checkpoint_file
+        step = model.restore(path_to_restore_checkpoint_file)
+        scheduler.last_epoch = step
+        print('Model restored from file: %s' % path_to_restore_checkpoint_file)
+
+    path_to_losses_npy_file = os.path.join(path_to_log_dir, 'losses.npy')
+    if os.path.isfile(path_to_losses_npy_file):
+        losses = np.load(path_to_losses_npy_file)
+    else:
+        losses = np.empty([0], dtype=np.float32)
+
+    path_to_test_losses_npy_file = os.path.join(path_to_log_dir, 'test_losses.npy')
+    if os.path.isfile(path_to_test_losses_npy_file):
+        test_losses = np.load(path_to_test_losses_npy_file)
+    else:
+        test_losses = np.empty([0], dtype=np.float32)
+
+    while True:
+        for batch_idx, (images, length_labels, digits_labels, _) in enumerate(train_loader):
+            start_time = time.time()
+            images, length_labels, digits_labels = images.cpu(), length_labels.cpu(), [digit_labels.cpu() for digit_labels in digits_labels]
+            length_logits, digit1_logits, digit2_logits, digit3_logits, digit4_logits, digit5_logits = model.train()(images)
+            loss = _loss(length_logits, digit1_logits, digit2_logits, digit3_logits, digit4_logits, digit5_logits, length_labels, digits_labels)
+
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            scheduler.step()
+            step += 1
+            duration += time.time() - start_time
+
+            if step % num_steps_to_show_loss == 0:
+                examples_per_sec = batch_size * num_steps_to_show_loss / duration
+                duration = 0.0
+                print('=> %s: step %d, loss = %f, learning_rate = %f (%.1f examples/sec)' % (
+                    datetime.now(), step, loss.item(), scheduler.get_lr()[0], examples_per_sec))
+
+            if step % num_steps_to_check != 0:
+                continue
+
+            losses = np.append(losses, loss.item())
+            np.save(path_to_losses_npy_file, losses)
+
+            print('=> Evaluating on validation dataset...')
+            accuracy, test_loss_args = evaluator.evaluate(model)
+            test_loss = _loss(*test_loss_args, length_labels, digits_labels)
+            test_losses = np.append(test_losses, test_loss.item())
+            np.save(path_to_losses_npy_file, test_losses)
```

Comment:
Just a question,
so ur storing both the training and validation losses in the same .npy file

Reply:
nvm figured it out, it's 2 separate files dw

Reply:
actually I found something I missed, hold off from merging